lib/server/services/files.js

```diff
@@ -0,0 +1,135 @@
+'use strict';
+
+const errors = require('storj-service-error-types');
+const log = require('../../logger');
+
+const { DELETING_FILE_MESSAGE } = require('../queues/messageTypes');
+
+// eslint-disable-next-line complexity
+async function removeFile(
+  storage,
+  { idBucket, userEmail, idFile },
+  { beforePointerIsRemoved = async () => { } }) {
+  const { Bucket, BucketEntry, Frame, Pointer } = storage.models;
+
+  if (!Bucket || !BucketEntry || !Frame || !Pointer) {
+    throw new errors.InternalError('Missing required storage models');
+  }
+
+  if (!idBucket || !userEmail || !idFile) {
+    throw new errors.InternalError('Missing required params');
+  }
+
+  try {
+    const bucket = await Bucket.findOne({ _id: idBucket });
+
+    if (!bucket) {
+      throw new errors.NotFoundError('Bucket not found');
+    }
+
+    if (bucket.user !== userEmail) {
+      throw new errors.ForbiddenError();
+    }
+
+    const bucketEntry = await getBucketEntry(BucketEntry, { idBucket: bucket._id, idFile });
+
+    if (!bucketEntry) {
+      throw new errors.NotFoundError('File not found');
```

Comment:
`errors` [has HTTP errors](https://github.com/internxt/service-error-types/blob/master/index.js), which is in fact coupling the service to HTTP (and assuming everything is an HTTP error)